lib/subRequire.js

```diff
@@ -25,6 +25,15 @@ var modulePath = {}; //maintain a list of the paths where we resolved files - ne
  * but also provides a way to load submodules by passing the optional parentId argument.
  */
 module.exports = function (id, parentId) {
+    // load mock if appropriate
+    if (global.__mockServices &&
+        _.includes(global.__mockServices, path.basename(id, path.extname(id)))
+    ) {
+        var mockPath = path.resolve(global.__appDir, 'test', 'bos-mocks', 'services', path.basename(id));
+        if (mockPath) {
```

Comment:
on failure to find the service, would a message be helpful?
maybe like:
```javascript
logger.warn(util.format('the requested mock service "%s" was not found in the mock service directory (%s)', mockDir, mockId));
```
(assuming the mock directory has been calculated once into a variable named `mockDir` and `mockId = path.basename(id)` was done too ... `util.format` should be used for safety since users can configure different loggers)

Reply:
Good idea, definitely helpful.